Scheduled Updates: Add batch mutations

TimBroddin opened the pull request:
## Proposed Changes

This PR lays the groundwork for batch creating, updating & deleting scheduled updates across sites. Since the feature is behind a feature flag for now, I believe we can merge this (if there's no discussion on code design) and iterate later.

* Adds mutations for `useBatchCreateUpdateScheduleMutation`, `useBatchEditUpdateScheduleMutation` and `useDeleteUpdateScheduleMutation`
* Implement Save button (for create)

When running the mutations, an error array will be thrown for all encountered errors. 

TODO (in future PR's):

- Do something with the errors

## Exploration

This task started with an exploration into being able to do add a `useMultiSiteMutation(siteSlugs, mutation)` hook. The idea was that this hook would allow you to pass in an array of slugs and a mutation, which in turn would trigger a mutation for each site passed in.

```
const { mutate } = useMultiSiteMutation( siteSlugs,  useCreateUpdateScheduleMutation);

...
mutate(params)
```

However, this broke the [rules of hooks](https://legacy.reactjs.org/docs/hooks-rules.html). React expects the same amount of hooks to be called on every render. Since siteSlugs would be of variable length, a variable amount of mutation hooks would be created on every render and React would explode.

I tried some workaround, but they didn't feel like great design, so I resorted to creating new variations of the existing mutation hooks.

Some discussion on batching mutations: https://github.com/TanStack/query/discussions/4009 

## Testing Instructions

1. Apply this PR
2. Go to `http://calypso.localhost:3000/plugins/scheduled-updates/create?flags=plugins/multisite-scheduled-updates`
3. Create a schedule for multiple sites
4. Check in network inspector if multiple requests go out.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

TimBroddin committed:
Add batch mutations

ouikhuan reviewed (commented):
Nice work! Tested the create context and it does fires to the target sites I tested. However, when we create a schedule, we also need to check and see if a site's monitoring module is activated and has wp-cron.php added or not, something like what we did here: https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/plugins-scheduled-updates/hooks/use-create-monitor.tsx. Do you think it'll be difficult to add this after create schedules in the multi-site context? Or should we move this part and handle it in the backend?

TimBroddin committed:
Add batch monitor settings

TimBroddin commented:
Hey @ouikhuan 👋  I added batch support for the monitor settings.

ouikhuan commented on client/blocks/plugins-scheduled-updates-multisite/schedule-form.tsx:
```diff
@@ -60,31 +68,86 @@ export const ScheduleForm = () => {
+			createMonitors();
+			onNavBack && onNavBack();
+		} catch ( error ) {
+			createMonitors();
```
Do we need to call `createMonitors` when it's failing?

TimBroddin replied:
It could be succeeding for some. But you're right, we only need to call it for those that are failing.

ouikhuan commented on client/blocks/plugins-scheduled-updates/hooks/use-create-monitor.tsx:
```diff
@@ -18,33 +47,39 @@ export function useCreateMonitor( siteSlug: SiteSlug ) {
+	const siteUrls = useSelector( ( state ) =>
+		siteIds.map( ( siteId ) => getSiteUrl( state, siteId as number ) )
+	);
+	const { createMonitorSettings } = useBatchCreateMonitorSettingsMutation( siteSlugs );
```
Do we need to pass the `siteSlugs` as a params in `useBatchCreateMonitorSettingsMutation` function here?

TimBroddin replied:
I was wrestling a lot with [the rules of hooks](https://legacy.reactjs.org/docs/hooks-rules.html) yesterday. I'm going to refactor this, I thought of a better way 😀

TimBroddin replied:
You're right, not needed!

TimBroddin committed:
- Only create schedules for sites that have succeeded
- Simplify useCreateMonitors() - move siteSlugs parameter to returned function
- Simplify useBatchCreateUpdateScheduleMutation - now returns errors instead of throwing them. This simplifies the error handling.

ouikhuan reviewed (approved):
Great work, thanks for addressing all the feedback 👍
